DoSomething/gambit-campaigns
# textPostConfig and photoPostConfig templates [pr]

#### What's this PR do?

Returns the field values from a botConfig's `postConfig` entry as properties in the `templates` object of a `GET /campaigns/:id` response.

Contentful changes:

*  [x] Added a new `photoPostConfig` content type, with `photoPost` fields like `gambitSignupMenuMessage`, `askQuantityMessage`

* [x] Added `askTextMessage`, `invalidTextMessage`, and `menuCompletedMessages` fields to the `textPostConfig` content type.

* [x] Added `photoPostConfig` as an available type to reference as a botConfig's `postConfig` field.

* [x] Create `photoPostConfig` entries for all botConfigs currently running on staging and production, migrating overrides. This sounds like the stuff Contentful migrations are made for, but the number's small enough that I'll just manually add 💻 📝 Campaigns id's to edit:

    * [x] 7 - Mirror Messages
    * [x] 2178  - Give A Spit (thirdParty)
    * [x] 7059 - VCard (thirdParty)
    * [x] 7483 - Rinse Recycle Repeat
    * [x] 7951 - Gun Violence (thirdParty)
    * [x] 8017 - Grab the Mic (thirdParty)
    * [x] 8038 - Save Mascots
<br>

* [x] Make `postConfig` field mandatory 

After go live:

* [ ] Remove botConfig fields like `askQuantity` that have been migrated to the photoPostConfig type.

Gambit changes:

* Renames the botConfig `config.templates` as `config.templatesByContentType`, adds `textPostConfig` and `photoPostConfig` properties (#1032) per Contentful changes above 👆 

* Refactors the `botConfig` helper to gather templates based on content type

* Adds new `askText` and `invalidText` replies for use in Text Post middleware

#### How should this be reviewed?

Upon staging deploy:

* Verify textPost conversation flow (LUCKYBOT)

* Verify photoPost conversation flow (BOOKBOT)

#### Relevant tickets
#1021 

#### Checklist
- [x] Deploy https://github.com/DoSomething/gambit-conversations/pull/317 to Conversations prod before deploying this branch to Campaigns prod
